### ISSUE TITLE
[AccessibilityInsight] Sometimes, the focus rectangle focus on property's edit part when using up/down to switch property items in PropertyGrid

### ISSUE THREAD

**Issue by MelonWang1** [dotnet/winforms#12639]
### .NET version

.NET 10.0.100-alpha.1.24611.5

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, cannot repro in .NET 8.0. Also repro in .NET 9.0 + binaries built from release/9.0 branch of winforms repro. And caused by PR[12479](https://github.com/dotnet/winforms/pull/12479).

### Issue description

Sometimes, the focus rectangle focus on property's edit part when using up/down to switch property items in PropertyGrid .

**Actual result:** 

The focus rectangle sometimes focus to edit part. 
![Image](https://github.com/user-attachments/assets/2708df5e-e23a-4dd0-abbc-ee473a59d14f)

https://github.com/user-attachments/assets/2e4960a8-bcf5-4307-8fdd-529d3b38aeef

**Expected result:** 

The focus rectangle always focus to property item. 

https://github.com/user-attachments/assets/e581045c-aeb9-4bc8-9c4a-275293f88037

**More info:** 

1. This issue cannot repro in Inspect/Narrtor/NVDA tool. 

2. This issue also repro in go through the drop down list item using up/down keyboard. 
 ![Image](https://github.com/user-attachments/assets/f6e97ad6-5f98-4cd2-b411-f492a53aa890)

### Steps to reproduce

1. Create a .NET Core Winforms project. 
2. Add a TextBox and PropertyGrid controls in form designer. 
3. Set the SelectedObject property to TextBox1 in propertyGrid1 control. 
4. Build and run project. 
5. Open Accessible Insight tool and select anyone property in propertyGrid control. 
6. Press up/down keyboard.

**Comment by Tanya-Solyanik** [comment]
Closing because this works with the main accessibility tools and is broken on ly in a testing tool -AccessibilityInsights.
